### PR TITLE
Upgrade git to version 2.10.1, plus a fix to uninstall_preflight.

### PR DESCRIPTION
**Pull request by athrunsun**
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed

### CODE REVIEW

File: Casks/git.rb [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
   else
-    version '2.7.1'
-    sha256 '260b32e8877eb72d07807b26163aeec42e2d98c350f32051ab1ff0cc33626440'
+    version '2.10.1'
+    sha256 '6bad786bae098eaae0aa824e714d5a0b0fb3eb5d1412ed27c89743ddc1d3d209'
 
     url "https://downloads.sourceforge.net/git-osx-installer/git-#{version}-intel-universal-mavericks.dmg"
     appcast 'https://sourceforge.net/projects/git-osx-installer/rss',
-            checkpoint: '96f8b3facd8324ed75a2dc919d33519506b892f2b3bd8a84d16f2a6f93c9f6a5'
+            checkpoint: '56ba7834c0c58ff1b9f9ab9c7660d7a3228339e6114224a665b0285c5cc5d800'
```
Comment (athrunsun) [inline Casks/git.rb]: This is calculated with `curl --compressed --location --user-agent 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.152 Safari/537.36' "https://sourceforge.net/projects/git-osx-installer/rss" | /usr/bin/sed 's|<pubDate>[^<]*</pubDate>||g' | shasum --algorithm 256` as instructed in [stanza appcast](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md).

File: Casks/git.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
   homepage 'https://sourceforge.net/projects/git-osx-installer/'
 
   uninstall_preflight do
-    system "/usr/bin/yes yes | #{staged_path}/uninstall.sh"
+    system '/usr/bin/yes yes | /usr/local/git/uninstall.sh'
```
Comment (athrunsun) [inline Casks/git.rb]: Git installs into `/usr/local/git` as instructed in readme in its `.dmg`. `#{staged_path}` points to `/usr/local/Caskroom/git/2.10.1/` thus invalid.

I tried `#{prefix}/uninstall.sh` but I'm told `Warning: Unexpected method 'prefix' called during uninstall_preflight on Cask git.` Is there any better way to represent `/usr/local/git`?

Comment (miccal) [inline Casks/git.rb]: I don't think so since it installs a `pkg` not an `app`.

**Review by miccal — APPROVED** [review]

File: Casks/git.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@
   homepage 'https://sourceforge.net/projects/git-osx-installer/'
 
   uninstall_preflight do
-    system '/usr/bin/yes yes | /usr/local/git/uninstall.sh'
+    system "/usr/bin/yes yes | #{staged_path}/uninstall.sh"
```
Comment (athrunsun) [inline Casks/git.rb]: @vitorgalvao I'm sorry but why is this reverted to `#{staged_path}`? git-osx-installer is no longer installing its binaries to that location.

Comment (miccal) [inline Casks/git.rb]: @athrunsun just tested and you are correct - I will fix this.

Comment (athrunsun) [inline Casks/git.rb]: @miccal OK, thx.

Comment (vitorgalvao) [inline Casks/git.rb]: @athrunsun My mistake, I thought the script came in the dmg. If it didn’t, then this line was never correct.

**Pull request merged**